typedefs.h

```diff
@@ -38,6 +39,16 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #endif
 #endif
 
+// OpenStreetMap node ids are higher than 2^32
+OSRM_STRONG_TYPEDEF(uint64_t, NodeID64)
+OSRM_STRONG_TYPEDEF(uint32_t, OSMWayID)
```

Comment:
should be fine

http://wiki.openstreetmap.org/wiki/Stats#Nodes.2C_ways_and_relations